### PR TITLE
Upgrade tensorflow to version to 2.6.x

### PR DESCRIPTION
**Pull request by ophirfrish**
Upgrade the following versions:
keras - from 2.4.3 to 2.6
tensorflow - from 2.4.2 to 2.6.2
h5py - from version < 3.0 to version 3.1.0

cc @areusch @leandron @Mousius @gromero for reviews

Co-authored-by:  Chris Sidebottom Chris.Sidebottom@arm.com

### CODE REVIEW

File: docker/install/ubuntu_install_tensorflow.sh [hunk 1]
```diff
@@ -21,6 +21,5 @@ set -u
 set -o pipefail
 
 pip3 install \
-    "h5py<3.0" \
```
Comment (lhutton1) [inline docker/install/ubuntu_install_tensorflow.sh:24]: It would also be good to increment the version of `h5py`, otherwise installing tensorflow gives:
```
ERROR: tensorflow 2.6.2 has requirement h5py~=3.1.0, but you'll have h5py 2.10.0 which is incompatible.
```

**Review by lhutton1 — COMMENTED** [review]
Thanks @ophirfrish! Just a small thing below

**Review by lhutton1 — APPROVED** [review]
LGTM!

**Comment by leandron** [comment]
I'm using this PR to generate a set of temporary images so that we can validate the proposed change: https://ci.tlcpack.ai/blue/organizations/jenkins/docker-images-ci%2Fdaily-docker-image-rebuild/detail/daily-docker-image-rebuild/204/pipeline

**Comment by leandron** [comment]
New attempt to generate docker images https://ci.tlcpack.ai/blue/organizations/jenkins/docker-images-ci%2Fdaily-docker-image-rebuild/detail/daily-docker-image-rebuild/208/pipeline, after fixing https://github.com/tlc-pack/tlcpack/pull/87.

**Comment by leandron** [comment]
@jtuyls @anilmartha, in the process of updating TF to 2.6 we (@ophirfrish @Mousius and me) bumped into an issue that `cpptests` were crashing, which turns out to be some conflicting `h5py` (versions `2.x` and `3.x`) installations in some Vitis related scripts. We are proposing to unify the `h5py` installation to be done only in the TF script.

Can you please review and in case you disagree, propose some options to those changes?

**Comment by jtuyls** [comment]
> @jtuyls @anilmartha, in the process of updating TF to 2.6 we (@ophirfrish @Mousius and me) bumped into an issue that `cpptests` were crashing, which turns out to be some conflicting `h5py` (versions `2.x` and `3.x`) installations in some Vitis related scripts. We are proposing to unify the `h5py` installation to be done only in the TF script.
> 
> Can you please review and in case you disagree, propose some options to those changes?

@leandron Could we keep the h5py being installed separately in the vitis_ai_core.sh script: https://github.com/apache/tvm/pull/10084/files#diff-7773869aa5e34502bbaa476fe3339c2e3956199b0be759084feee12fe8bcdfe1L37?
In our Dockerfile we don't use the ubuntu_install_tensorflow.sh script so h5py won't be installed otherwise. This shouldn't affect the CI as this script is not used there. 
The changes in vitis_ai_packages_ci script are fine for us.

**Comment by leandron** [comment]
> @leandron Could we keep the h5py being installed separately in the vitis_ai_core.sh ?

And with regards to versions, do you want to keep it as is, with version 2.x or update to be coherent with the one installed in TVM, which means h5py 3.x ?

**Comment by jtuyls** [comment]
> > @leandron Could we keep the h5py being installed separately in the vitis_ai_core.sh ?
> 
> Sure, and with regards to versions, do you want to keep it as is, with version 2.x or update to be coherent with the one installed in TVM, which means h5py 3.x ?

Yeah, I think it's best to keep it as is (2.10.0) for now and I can work on updating it later to 3.x. I think 3.x works but I would like to completely verify that first.

**Comment by leandron** [comment]
After the last changes, I'm rebuilding the Docker images again to re-run tests - https://ci.tlcpack.ai/blue/organizations/jenkins/docker-images-ci%2Fdaily-docker-image-rebuild/detail/daily-docker-image-rebuild/230/pipeline/

**Comment by leandron** [comment]
With the latest version of this PR, rebased on top of main at ``, and with #10406 applied, I generated a new set of Docker images for testing at https://ci.tlcpack.ai/job/docker-images-ci/job/daily-docker-image-rebuild/238/, with label `tlcpackstaging/ci_****:20220301-163838-977bdd3d2`.

**Comment by leandron** [comment]
This is validated in https://ci.tlcpack.ai/blue/organizations/jenkins/tvm/detail/ci-docker-staging/232/pipeline/, covering up to 479785099.

@ophirfrish can you trigger CI again, so that I can merge this?

**Comment by leandron** [comment]
This is merged now, as it is running in production CI starting on `tlcpack/ci-gpu:v0.82` and `tlcpack/ci-cpu:v0.82`.

Thanks @ophirfrish @lhutton1 and @jtuyls!
